src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -390,14 +390,15 @@ def load(self,
 
         # Previous versions incorrectly stored the global config in
         # the localized module option. Check the config is here and fix it.
-        if not schedule_cfg:
-            schedule_cfg = self.handler.module.get_localized_module_option(
-                self.handler.MODULE_OPTION_NAME, '')
-            if schedule_cfg:
```

Comment:
I suppose moving the code below under `if not schedule_cfg:` block should fix the issue you described.

Reply:
I thought of this too. But I don't think this will result in the local module option being detected and removed when both the global module option and the local module option exist. You described this situation here,
https://github.com/ceph/ceph/pull/37864#discussion_r513533495
_I intentionally put this out of the if clause, so it always had a chance to clean the old option. Consider a case when an old localized option exists for both mgr.x and mgr.y (mgr/rbd_support/x/mirror_snapshot_schedule and mgr/rbd_support/y/mirror_snapshot_schedule). After upgrade the active mgr.x "fixes" mgr/rbd_support/x/mirror_snapshot_schedule making it mgr/rbd_support/mirror_snapshot_schedule. Then mgr.y becomes active, and if it is under if clause it will not remove mgr/rbd_support/y/mirror_snapshot_schedule because mgr/rbd_support/mirror_snapshot_schedule exists. And then if the user decides to remove the global schedule (mgr/rbd_support/mirror_snapshot_schedule) the mgr.y will mysteriously reuse mgr/rbd_support/y/mirror_snapshot_schedule_

Reply:
Indeed. Still, taking how unlikely one will face this situation now after we have been have this "remove localized option" code for more than two years (and actually about to remove it as unnecessary any more), especially taking into the consideration that the bug was introduced at the time when the feature was new and not many users used it,  I vote for this `if not schedule_cfg`  solution, unless someone see a better one. Fixing the audit log flood looks more important to me.

Reply:
I second the "do it on init" approach.  Then `load` could always query just the global option, making the common code path more straightforward.

Reply:
Done. Thanks!